bn_mp_grow.c

```diff
@@ -29,7 +29,7 @@ int mp_grow(mp_int *a, int size)
        * in case the operation failed we don't want
        * to overwrite the dp member of a.
        */
-      tmp = (mp_digit *) XREALLOC(a->dp, sizeof(mp_digit) * (size_t)size);
+      tmp = (mp_digit *) XREALLOC(a->dp, (size_t)size * sizeof(mp_digit));
```

Comment:
Wouldn't it be better here to use "(unsigned)size" in stead of "(size_t)size"?  Maybe some processor has a 32bit multiplyer while still size_t is 64bit.   That more accurately describes what's the minimum expectation here.

Reply:
It does not matter. Both factors are size_t and size_t is the expected argument. The multiplication should also be done in size_t for overflow avoidance. If you use unsigned it would also work due to promotion. But I think it is less unclear since you involve a third type (int, unsigned and size_t).

Reply:
I'm OK with it, anyway!